backend/api/services.py

```diff
@@ -173,3 +237,153 @@ def get_reddit_posts(subreddit, symb, time="week") -> pd.DataFrame:
     df.replace("", np.nan, inplace=True)
     df.dropna(inplace=True)
     return df
+
+
+# TWEETS
+def get_tweets(query) -> pd.DataFrame:
+    """Returns a dataframe of tweets matching a given search query.
+
+    Args:
+        query (str): Search query eg. $AAPL
+
+    Returns:
+        pd.DataFrame: DataFrame of tweets and other relevant information.
+    """
+    query = "\$" + query + " lang:en"
+    tweets = twitter.search_recent_tweets(
+        query=query, tweet_fields=["author_id", "created_at", "public_metrics", "entities"], max_results=100
+    )
+
+    tweets_df = pd.DataFrame(tweets.data)
+
+    # explode out public metrics
+    pub_metrics = tweets_df["public_metrics"].apply(pd.Series)
+
+    # extract hashtags
+    entities = pd.json_normalize(tweets_df["entities"])
+    hashtags = []
+    for i in entities["hashtags"]:
+        hashtags_str = ""
+        if isinstance(i, list):
+            for j in i:
+                hashtags_str += j["tag"] + " "
+        hashtags.append(hashtags_str)
+    hashtags = pd.Series(hashtags).rename("hashtags")
+
+    tweets_df = pd.concat([tweets_df, hashtags, pub_metrics], axis=1)
+    tweets_df.drop(columns=["public_metrics", "edit_history_tweet_ids", "entities"], inplace=True)
+    tweets_df["text"] = tweets_df["text"].apply(cleanTxt)
+
+    # publicity score by summing pub metrics
+    tweets_df["pub_score"] = tweets_df[["retweet_count", "reply_count", "like_count", "quote_count"]].sum(axis=1)
+    tweets_df["sentiment"] = tweets_df["text"].apply(lambda x: get_sentiment(x))
```

Comment:
`apply` expects a function so passing it as `.apply(get_sentiment)` is preferrable and more readable.

Reply:
thanks!!